dask/dataframe/tests/test_dataframe.py

```diff
@@ -4135,6 +4135,24 @@ def test_set_index_with_index():
     assert ddf2 is ddf
 
 
+def test_set_index_no_sort():
```

Comment:
Let's move this test to `dask/dataframe/tests/test_shuffle.py` with the other `set_index`-related tests

Reply:
https://github.com/dask/dask/pull/10342/commits/81e4df931b12dc0beda219d841e27673bf4a4fbf